sky/packages/sky/lib/src/widgets/transitions.dart

```diff
@@ -6,117 +6,91 @@ import 'package:sky/animation.dart';
 import 'package:sky/src/widgets/animated_component.dart';
 import 'package:sky/src/widgets/basic.dart';
 import 'package:sky/src/widgets/framework.dart';
+import 'package:sky/src/widgets/global_key_watcher.dart';
 import 'package:vector_math/vector_math.dart';
 
 export 'package:sky/animation.dart' show Direction;
 
-// A helper class to anchor widgets to one another. Pass an instance of this to
-// a Transition, then use the build() method to create a child with the same
-// transition applied.
-class Anchor {
-  Anchor();
+class TransitionProxy extends GlobalKeyWatcher {
 
-  TransitionBase transition;
-
-  Widget build(Widget child) {
-    return new _AnchorTransition(anchoredTo: this, child: child);
-  }
-}
-
-// Used with the Anchor class to apply a transition to multiple children.
-class _AnchorTransition extends AnimatedComponent {
-  _AnchorTransition({
+  TransitionProxy({
     Key key,
-    this.anchoredTo,
+    GlobalKey transitionKey,
     this.child
-  }) : super(key: key);
+  }) : super(key: key, watchedKey: transitionKey);
 
-  Anchor anchoredTo;
   Widget child;
-  TransitionBase get transition => anchoredTo.transition;
-
-  void initState() {
-    if (transition != null)
-      watch(transition.performance);
-  }
 
-  void syncConstructorArguments(_AnchorTransition source) {
-    if (transition != null && isWatching(transition.performance))
-      unwatch(transition.performance);
-    anchoredTo = source.anchoredTo;
-    if (transition != null)
-      watch(transition.performance);
+  void syncConstructorArguments(TransitionBase source) {
     child = source.child;
     super.syncConstructorArguments(source);
   }
 
+  bool debugValidateWatchedWidget(Widget candidate) {
+    return candidate is TransitionBaseWithChild;
+  }
+
   Widget build() {
-    if (transition == null)
-      return child;
-    return transition.buildWithChild(child);
+    if (watchedWidget is TransitionBaseWithChild)
+      return watchedWidget.buildWithChild(child);
+    return child;
```

Comment:
Should this assert if it's not a TransitionBaseWithChild? I'm wondering if that should be valid.